pkg/drivers/host_driver.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"encoding/json"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"bytes"
+
+	"github.com/GoogleCloudPlatform/container-structure-test/pkg/types/unversioned"
+)
+
+type HostDriver struct {
+	ConfigPath string // path to image metadata config on host fs
+	GlobalVars []unversioned.EnvVar
+}
+
+func NewHostDriver(args DriverConfig) (Driver, error) {
+	return &HostDriver{
+		ConfigPath: args.Metadata,
+	}, nil
+}
+
+func (d *HostDriver) Destroy() {
+	// since we're running on the host, don't do anything
+}
+
+func (d *HostDriver) Setup(envVars []unversioned.EnvVar, fullCommands [][]string) {
+	// since we're running on the host, we'll provide an optional teardown field for`
+	// each test that will allow users to undo the setup they did.
+	// keep track of the original env vars so we can reset later.
+	d.GlobalVars = SetEnvVars(envVars)
+	for _, cmd := range fullCommands {
+		d.ProcessCommand(nil, cmd)
+	}
+}
+
+func (d *HostDriver) Teardown(envVars []unversioned.EnvVar, fullCommands [][]string) {
+	// since we're running on the host, we'll provide an optional teardown field for each test that
+	// will allow users to undo the setup they did.
+	ResetEnvVars(d.GlobalVars)
+	for _, cmd := range fullCommands {
+		d.ProcessCommand(nil, cmd)
+	}
+}
+
+// given a list of environment variable key/value pairs, set these in the current environment.
+// also, keep track of the previous values of these vars to reset after test execution.
+func SetEnvVars(vars []unversioned.EnvVar) []unversioned.EnvVar {
+	var originalVars []unversioned.EnvVar
+	for _, env_var := range vars {
+		originalVars = append(originalVars, unversioned.EnvVar{env_var.Key, os.Getenv(env_var.Key)})
+		if err := os.Setenv(env_var.Key, os.ExpandEnv(env_var.Value)); err != nil {
+			logrus.Fatalf("error setting env var: %s", err)
+		}
+	}
+	return originalVars
```

Comment:
How about this just return a function to reset them?

https://github.com/GoogleCloudPlatform/skaffold/blob/2541b68a29f4141ce35200c27e1ca59d7bc4a62b/testutil/util.go#L82-L102